### ISSUE TITLE
Feature request: Support multiple messages in messages array (for chat.openai.com)

### ISSUE THREAD

**Issue by acheong08** [linweiyuan/go-chatgpt-api#59]
It throws a 422 error when using https://github.com/acheong08/ChatGPT-to-API (which works when directly accessing chat.openai.com or alternative proxies.)

Thank you

**Comment by linweiyuan** [comment]
can share the requestBody? should be something wrong with this

if 422, an `errorMessage` is also returned, you can check it for more details

**Comment by acheong08** [comment]
```go
package types

import "github.com/google/uuid"

type chatgpt_message struct {
	ID      uuid.UUID       `json:"id"`
	Role    string          `json:"role"`
	Content chatgpt_content `json:"content"`
}

type chatgpt_content struct {
	ContentType string   `json:"content_type"`
	Parts       []string `json:"parts"`
}

type ChatGPTRequest struct {
	Action          string            `json:"action"`
	Messages        []chatgpt_message `json:"messages"`
	ParentMessageID string            `json:"parent_message_id,omitempty"`
	Model           string            `json:"model"`
}

func NewChatGPTRequest() ChatGPTRequest {
	return ChatGPTRequest{
		Action:          "next",
		ParentMessageID: uuid.NewString(),
		Model:           "text-davinci-002-render-sha",
	}
}

func (c *ChatGPTRequest) AddMessage(role string, content string) {
	c.Messages = append(c.Messages, chatgpt_message{
		ID:      uuid.New(),
		Role:    role,
		Content: chatgpt_content{ContentType: "text", Parts: []string{content}},
	})
}
```

It's a lot more minimal

**Comment by acheong08** [comment]
```json
{
	"errorMessage": "body,messages,1,author,role -> value is not a valid enumeration member; permitted: 'unknown', 'user', 'assistant', 'system', 'critic', 'discriminator', 'tool'"
}
```

**Comment by acheong08** [comment]
`https://ai.fakeopen.com/api/conversation` works just fine.

Request:
```json
{
	"action": "next",
	"messages": [
		{
			"id": "cd465ab7-3ee4-40e7-8e48-ade9926ad68e",
			"role": "user",
			"content": {
				"content_type": "text",
				"parts": [
					"What is 1+1?"
				]
			}
		},
		{
			"id": "cd465ab7-3ee4-40e7-8e48-ade9926ad68f",
			"role": "assistant",
			"content": {
				"content_type": "text",
				"parts": [
					"3"
				]
			}
		}
	],
	"parent_message_id": "572aca1b-59e5-4262-85b6-b258fa5a38b8",
	"model": "text-davinci-002-sha"
}
```

**Comment by acheong08** [comment]
It only happens when there are multiple elements in messages array

**Comment by linweiyuan** [comment]
IC, let me test it tonight and find a way to support this format

**Comment by acheong08** [comment]
hmm let me try again

**Comment by acheong08** [comment]
I'm still getting the same error
